src/dmd/identifier.d

```diff
@@ -124,6 +125,13 @@ nothrow:
 
     extern (C++) static __gshared StringTable stringtable;
 
+    /**
+       A secondary string table is used to guarantee that we generate unique
+       identifiers per module. See generateIdWithLoc and issues
+       16995, 18097, 18111, 18880, 18868, 19058
+     */
+    private extern (C++) static __gshared StringTable fullPathStringTable;
```

Comment:
If this is `private` there shouldn't be a reason for it being `extern(C++)`.

Reply:
I don't know why the other one is. Maybe because of the backend. Speaking of which, I need to add this to the .h file.

I'm following the existing pattern.

Reply:
I recommend not following existing patterns.  We need to get rid of this cruft somehow.

Reply:
There are header files here for a reason. Unless they disappear, I think `extern(C++)` is the way to go.

Reply:
`static` and `__gshared` are redundant. Dscanner which is in master would have complained about this.